test/units/executor/test_action_write_locks.py

```diff
@@ -0,0 +1,99 @@
+# (c) 2021, Kevin Carter <kecarter@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+
+from multiprocessing import Process
+
+from multiprocessing.managers import DictProxy
+
+import pytest
+
+from ansible.executor import action_write_locks
+from ansible.module_utils import six
+
+
+ACTION_LOCK = action_write_locks.action_lock
+ACTION_WRITE_LOCKS = action_write_locks.action_write_locks
+
+
+def _run_threads(test_int):
```

Comment:
This name seems misleading. Maybe use something along the lines of "acquire an in-thread lock"?

Reply:
done.